app-android/src/androidTest/java/com/ustadmobile/port/android/screen/ReportEditScreen.kt

```diff
@@ -1,72 +1,99 @@
+    val dateRangeTextInput = KTextInputLayout { withId(R.id.fragment_edit_report_dialog_daterange_textinputlayout) }
 
-    val fromDateTextInput = KTextInputLayout { withId(R.id.activity_report_edit_fromDate_textinputlayout)}
+    val seriesRecycler: KRecyclerView = KRecyclerView({
+        withId(R.id.activity_report_edit_series_list)
+    }, itemTypeBuilder = {
+        itemType(::Series)
+    })
 
-    val toDateTextInput = KTextInputLayout { withId(R.id.activity_report_edit_toDate_textinputlayout)}
 
-    val personAddList = KTextView {
-        withId(R.id.item_createnew_line1_text)
-        withText("Add person filter")
+    class Series(parent: Matcher<View>) : KRecyclerItem<Series>(parent) {
+
+        val seriesNameTextInput = KTextInputLayout(parent) { withId(R.id.item_report_series_title_text_inputlayout) }
+
+        val seriesNameTextView = KTextView(parent) { withId(R.id.item_report_series_title_text) }
+
+        val deleteButton = KImageView(parent) { withId(R.id.item_report_series_delete_button) }
+
+        val dataSetTextInput = KTextInputLayout(parent) { withId(R.id.item_edit_report_dialog_yaxis_textinputlayout) }
+
+        val dataSetTextView = KTextView(parent) { withId(R.id.item_edit_report_dialog_yaxis_text) }
+
+        val visualTypeTextInput = KTextInputLayout(parent) { withId(R.id.item_edit_report_dialog_visual_type_textinputlayout) }
+
+        val visualTypeTextView = KTextView(parent) { withId(R.id.item_edit_report_dialog_visual_type_text) }
+
+        val subgroupTextInput = KTextInputLayout(parent) { withId(R.id.item_edit_report_dialog_subgroup_textinputlayout) }
+
+        val subgroupTextView = KTextView(parent) { withId(R.id.item_edit_report_dialog_subgroup_text) }
+
+        val filterRecycler = KRecyclerView(parent, { withId(R.id.item_report_edit_filter_list) }, { itemType(::Filter) })
+
+        val addFilterButton = KView(parent) { withId(R.id.item_edit_report_filter_add_layout)}
+
     }
 
-    val verbAddList = KTextView {
-        withId(R.id.item_createnew_line1_text)
-        withText("Add verb filter")
+    class Filter(parent: Matcher<View>) : KRecyclerItem<Filter>(parent) {
+
+        val filterName = KTextView(parent) { withId(R.id.item_clazz_simple_line1_text) }
+
+        val filterDeleteButton = KImageView(parent) { withId(R.id.item_clazz_simple_secondary_menu_imageview) }
     }
 
+
     val nestScroll = KNestedScrollView {
         withId(R.id.fragment_report_edit_edit_scroll)
     }
 
     fun fillFields(fragmentScenario: FragmentScenario<ReportEditFragment>? = null,
-                   report: ReportWithFilters,
-                   reportOnForm: ReportWithFilters? = ReportWithFilters(),
-                   setFieldsRequiringNavigation: Boolean = true, person: Person? = null,
-                   verbDisplay: VerbDisplay? = null, entry: ContentEntry? = null,
+                   updatedReport: ReportWithSeriesWithFilters,
```

Comment:
Let's avoid using the fillFields function for new code now. I think now that we have Kaspresso there it's not really that helpful, and it makes things less clear than just using the normal Kaspresso functions.

Reply:
@SamihMustafa  can you please also remove fillFields from the screen template?